src/xrCore/_math.cpp

```diff
@@ -195,6 +201,50 @@ XRCORE_API u64 GetCLK()
 
 bool g_initialize_cpu_called = false;
 
+
+uint32_t cpufreq()
+{
+    uint32_t cpuFreq = 0;
+
+    // CPU frequency is stored in /proc/cpuinfo in lines beginning with "cpu MHz"
+    std::string pattern = "^cpu MHz\\s*:\\s*(\\d+)";
+    const char* pcreErrorStr = NULL;
+    int pcreErrorOffset = 0;
+
+    pcre* reCompiled = pcre_compile(pattern.c_str(), PCRE_ANCHORED, &pcreErrorStr, &pcreErrorOffset, NULL);
+    if(reCompiled == NULL)
+    {
+        return 0;
+    }
+
+    std::ifstream ifs("/proc/cpuinfo");
+    if(ifs.is_open())
+    {
+        std::string line;
+        int results[10];
+
+        while(ifs.good())
+        {
+            getline(ifs, line);
+            int rc = pcre_exec(reCompiled, 0, line.c_str(), line.length(), 0, 0, results, 10);
+            if(rc < 0)
```

Comment:
I'd prefer to see inversed condition: 
```
if (rc >= 0) 
{
...
break;
}
```
It makes the code easier to read.

Reply:
Not agree. Small blocks always more readable. Instead of one big true statement (now 6-7 lines, but who knows, maybe in the future it'll be ten's of them), we have clean solution of just 1 row. It's really more readable. We look at the code, see that this condition is true (rc less than 0), see continue and we don't interesting in the rest code. Otherwise, we should look for the closing bracket. Bad for programmer.